2023 updates to orientation chapter. (#129)
moorepants/learn-multibody-dynamics: orientation.rst

oriented about the common parallel :math:`z` unit vectors of the two frames. We
then say "reference frame :math:`A` is oriented with respect to reference frame
:math:`N` about the shared :math:`z` unit vectors through an angle

> :math:`\theta`. A visual representation of this orientation looking into the
> :math:`z` unit vectors is:

Reviewer: ```suggestion
:math:`\theta`. A visual representation of this orientation looking from the direction of the positive :math:`z` unit vector is:
```

In any case, it should be vector (singular).

   A_C_N.transpose()

Notably, the inverse and the transpose are the same here. This indicates that

> this matrix is an `orthogonal matrix`_. All matrices that describe the

Reviewer: special orthogonal matrix would be complete

```suggestion
this matrix is a `special orthogonal matrix`_. All matrices that describe the
```

orientation between reference frames are orthogonal matrices. Following the
notation convention, this holds:

.. math::
   :label: dcm-inverse-transpose

   {}^N\mathbf{C}^A = \left({}^A\mathbf{C}^N\right)^{-1} = \left({}^A\mathbf{C}^N\right)^T

.. _orthogonal matrix: https://en.wikipedia.org/wiki/Orthogonal_matrix

.. admonition:: Exercise

   Write :math:`{}^N\mathbf{C}^A` for simple rotations about both the shared
   :math:`\hat{n}_x` and :math:`\hat{a}_x` and shared :math:`\hat{n}_y` and
   :math:`\hat{a}_y` axes, rotating :math:`A` with respect to :math:`N` through

>    angle :math:`\theta`.

Reviewer: The direction of the angle is never specified. Maybe the default direction is best explained at the spot where also the "right-handedness" of the frames is specified?

.. admonition:: Solution
   :class: dropdown

   For a :math:`x` orientation:

   .. math::

      \begin{bmatrix}
        1 &  0  & 0 \\

>         0 & \cos{\theta} & \sin{\theta} \\

Reviewer: It's early morning, but I have convinced myself that this is the inverse of the correct solution

Author: I think it is correct, at least per my definition of the rotation matrix notation.

Reviewer: Yes, it is a change in notation that tripped me up

   Three angles relating :math:`\hat{a}_x` to the unit vectors of :math:`N`.

Similar to the simple example above, we can write these equations if the
:math:`\alpha_y` and :math:`\alpha_z` angles relate the :math:`\hat{a}_y` and

> :math:`\hat{a}_z` unit vectors to those of :math:`N`:

Reviewer: Easiest to add this comment here rather than make new PR:
In the sentence "Since we are working with mutually perpendicular unit vectors ...", 'mutually perpendicular' and the second 'unit' are not needed, which might confuse readers who do not know the dot product <-> cosine rule by heart.

     \hat{a}_z \cdot \hat{n}_x &\hat{a}_z \cdot \hat{n}_y & \hat{a}_z \cdot \hat{n}_z
   \end{bmatrix}

We call :math:`{}^A\mathbf{C}^N` the "direction cosine matrix" as a general
description of the relative orientation of two reference frames. This matrix
uniquely defines the relative orientation between reference frames :math:`N`
and :math:`A`, it is invertible, and its inverse is equal to the transpose, as

> shown above in the simple example. The direction cosine matrix found in the

Reviewer: ```suggestion
shown above in the simple example. The determinant of the matrix is also always 1, to ensure both associated frames are right-handed. The direction cosine matrix found in the
```

.. admonition:: Solution
   :class: dropdown

   .. jupyter-execute::

      beta = sm.symbols('beta')

      C = me.ReferenceFrame('C')
      D = me.ReferenceFrame('D')
      E = me.ReferenceFrame('E')

      D.orient_axis(C, beta, -C.y)

      D.dcm(C)

   .. jupyter-execute::

      E.orient_explicit(D, C.dcm(D))
      E.dcm(D)

Euler Angles
============

The camera stabilization gimbal_ shown in :numref:`camera-gimbal`  has three
`revolute joints`_ that orient the camera :math:`D` relative to the handgrip
frame :math:`A`.

.. _camera-gimbal:

.. figure:: https://objects-us-east-1.dream.io/mechmotum/orientation-camera-gimbal.png
   :align: center

   Four reference frames labeled on the Turnigy Pro Steady Hand Camera Gimbal.
   *Image copyright HobbyKing, used under fair use for educational purposes.*

If we introduce two additional auxiliary reference frames, :math:`B` and
:math:`C`, attached to the intermediate camera frame members, we can use three
successive simple orientations to go from :math:`A` to :math:`D`. We can
formulate the direction cosine matrices for the reference frames using the same
technique for the successive simple orientations shown in :ref:`Successive

> Orientations`, but now we will have three dimensional orientation between

Reviewer: A bit unclear, maybe like this: 

```suggestion
Orientations`, but now we will have a sequence of three rotations  between
```

> :math:`A` and :math:`D` allowing :math:`D` to be oriented in any direction

Reviewer: Not sure that it is obvious that three is the right number, so would split:
```suggestion
:math:`A` and :math:`D`. With three rotations, :math:`D` can be oriented in any direction
```

Author: I tried to reword this:

```rst
successive simple orientations to go from :math:`A` to :math:`D`. We can
formulate the direction cosine matrices for the reference frames using the same
technique for the successive simple orientations shown in :ref:`Successive
Orientations`, but now our sequence of three orientations will enable us to
orient :math:`D` in any way possible relative to :math:`A` in three dimensional
space.
```

   D.dcm(A)

> With these three successive orientations the camera can be oriented arbitrarily

Reviewer: Seems slightly redundant. Perhaps rotation is a better word here?

Author: I had a mixture of "rotation" and "orientation" last year when I first wrote this, but ended up changing everything to "orientation" to be consistent and to match the SymPy method names.

relative to the handgrip frame. These successive
:math:`z\textrm{-}x\textrm{-}y` orientations are a standard way of describing
the orientation of two reference frames and are referred to as `Euler Angles`_
[#]_.

.. _Euler Angles: https://en.wikipedia.org/wiki/Euler_angles

There are 12 valid sets of successive orientations that can arbitrarily orient
one reference frame with respect to another. We will also refer to these 12
possible orientation sets as "body fixed orientations". As we will soon see, a
rigid body and a reference frame are synonymous from an orientation perspective

> and each successive orientations rotates about a shared unit vector fixed in

Reviewer: typo: orientation singular

```suggestion
and each successive orientation rotates about a shared unit vector fixed in
```
